### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by MaksymKos**
[DEMO LINK](https://MaksymKos.github.io/js_2048_game/)

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,268 @@
...
+
+  updateScore();
+};
+
+function updateScore() {
+  gameScore.innerHTML = score;
+}
+
+function renderBoard() {
+  for (let rowIndex = 0; rowIndex < rows; rowIndex++) {
```
Comment (OleziO) [inline src/scripts/main.js:122]: Rename `rowIndex` to `r`

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,268 @@
...
+    for (let cellIndex = 0; cellIndex < cells; cellIndex++) {
```
Comment (OleziO) [inline src/scripts/main.js:123]: Rename `cellIndex` to `c`

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,268 @@
...
+  }
+}
+
+function move(row) {
+  const noZerosRow = row.filter(cell => cell !== 0);
+  const mergedRow = [];
+  let mergedValue = 0;
+
+  for (let i = 0; i < noZerosRow.length; i++) {
+    if (noZerosRow[i] === noZerosRow[i + 1]) {
```
Comment (OleziO) [inline src/scripts/main.js:138]: At the last iteration, `i + 1` will be out of arr indexes

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,268 @@
...
+  updateScore();
+  renderBoard();
+};
+
+function moveUp() {
+  isWinner();
+  isLoser();
+  moveScore = 0;
+
+  for (let columnIndex = 0; columnIndex < 4; columnIndex++) {
```
Comment (OleziO) [inline src/scripts/main.js:199]: Move 4 to constant variable

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,268 @@
...
+  appendTile();
+  score += moveScore;
+
+  updateScore();
+  renderBoard();
+};
+
+function keyHandler(keyEvent) {
+  switch (keyEvent.code) {
+    case 'ArrowLeft':
```
Comment (OleziO) [inline src/scripts/main.js:252]: Try to move all cases to the object.

**Review by etojeDenys — CHANGES_REQUESTED** [review]
https://github.com/mate-academy/js_2048_game/assets/70220667/3ea215b5-ea97-4cfe-ab48-09c5f11c2212

if there is no cell to merge, then you should not add a new one

**Review by sTorba24 — APPROVED** [review]

**Pull request closed without merging**